fixtures/workers-with-assets-and-http-based-service-bindings/tests/index.test.ts

```diff
@@ -0,0 +1,163 @@
+import { setTimeout } from "timers/promises";
+import { fetch } from "undici";
+import { afterAll, beforeAll, describe, it } from "vitest";
+import { runWranglerDev } from "../../shared/src/run-wrangler-long-lived";
+
+describe("[Workers + Assets] HTTP-based service bindings to a Worker with assets", () => {
+	describe("Workers running in separate wrangler dev sessions", () => {
+		let ipWorkerA: string,
+			portWorkerA: number,
+			stopWorkerA: (() => Promise<unknown>) | undefined;
+		let ipWorkerB: string,
+			portWorkerB: number,
+			stopWorkerB: (() => Promise<unknown>) | undefined,
+			getOutputWorkerB: () => string;
+
+		beforeAll(async () => {
+			({
+				ip: ipWorkerB,
+				port: portWorkerB,
+				stop: stopWorkerB,
+				getOutput: getOutputWorkerB,
+			} = await runWranglerDev(resolve(__dirname, "..", "workerB"), [
+				"--port=0",
+				"--inspector-port=0",
+				// we need this flag in order to be able to test Cron Triggers
+				// see https://developers.cloudflare.com/workers/examples/cron-trigger/#test-cron-triggers-using-wrangler
+				"--test-scheduled",
+			]));
+
+			({
+				ip: ipWorkerA,
+				port: portWorkerA,
+				stop: stopWorkerA,
+			} = await runWranglerDev(resolve(__dirname, "..", "workerA"), [
+				"--port=0",
+				"--inspector-port=0",
+			]));
+		});
+
+		afterAll(async () => {
+			await stopWorkerA?.();
+			await stopWorkerB?.();
+		});
+
+		/*
+		 * Assert that WorkerB works as intended.
+		 * This is mainly a sanity check that our test suite was configured correctly
+		 */
+		describe("WorkerB", () => {
+			it("should respond with static asset content", async ({ expect }) => {
+				let response = await fetch(
+					`http://${ipWorkerB}:${portWorkerB}/index.html`
+				);
+				let text = await response.text();
+				expect(response.status).toBe(200);
+				expect(text).toContain('This is an asset of "worker-🐝"');
+
+				response = await fetch(`http://${ipWorkerB}:${portWorkerB}/busy-bee`);
+				text = await response.text();
+				expect(response.status).toBe(200);
+				expect(text).toContain("All 🐝🐝🐝 are 🐝sy. Please come back later");
+			});
+
+			it("should return User Worker response", async ({ expect }) => {
+				let response = await fetch(
+					`http://${ipWorkerB}:${portWorkerB}/no-assets-at-this-path`
+				);
+				let text = await response.text();
+				expect(response.status).toBe(200);
+				expect(text).toContain("Hello from worker-🐝 fetch()");
+			});
+
+			it("should respond to cron trigger", async ({ expect }) => {
+				let response = await fetch(
+					`http://${ipWorkerB}:${portWorkerB}/__scheduled`
+				);
+				let text = await response.text();
+				expect(response.status).toBe(200);
+				expect(text).toContain("Ran scheduled event");
+
+				// add a timeout to allow stdout to update
+				await setTimeout(500);
+				expect(getOutputWorkerB()).toContain(
+					"Hello from worker-🐝 scheduled()"
+				);
+			});
+		});
+
+		/*
+		 * Assert that WorkerA works as intended
+		 */
+		describe("WorkerA", () => {
```

Comment:
to clarify, `workerA` makes all the requests we are testing for and returns **one** response that includes all those responses. But for the sake of keeping things readable and having these tests serve as docs as well, I have split the tests into separate, individual bits per expected response.

Reply:
if folks feel like this is confusing, pls call that out